src/dune_rules/melange_rules.ml

```diff
@@ -139,6 +133,21 @@ let add_rules_for_entries ~sctx ~dir ~expander ~dir_contents ~scope
         build_js ~dir ~loc:(Some loc) ~pkg_name ~module_system:mel.module_system
           ~dst_dir ~obj_dir ~sctx ~build_dir ~lib_deps_js_includes m)
   in
+  let* () =
+    match mel.alias with
+    | None -> Memo.return ()
+    | Some alias_name ->
+      let alias = Alias.make alias_name ~dir in
+      let* build_paths =
+        Memo.parallel_map
```

Comment:
Why do you need `Memo.parallel_map` here?. You can just collect all the files inside the fold.

Reply:
You're right. Fixed in https://github.com/ocaml/dune/pull/6401/commits/1d3557961bd2d4d814110e8462cf5558fd756700, I also refactored to reuse the list of modules instead of calculating it twice.